### PR TITLE
Create paleontographica-abteilung-b-palaeobotany-palaeophytology.csl

### PR DESCRIPTION
**Pull request by damnation333**
via https://forums.zotero.org/discussion/57867/style-request-palaeontographica
Adapted for journal, book and chapter. Didn't have access to the journal, but found what I needed from article previews regarding in-text citation.

Potentially Abteilung A of this journal follows the same style. Style guidelines are different, but I was not able to confirm with actual publications.

### CODE REVIEW

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    <link href="https://www.schweizerbart.de/journals/palb/instructions" rel="documentation"/>
+    <author>
+      <name>Patrick O'Brien</name>
+      <email>obrienpat86@gmail.com</email>
+    </author>
+    <category citation-format="author-date"/>
+    <category field="science"/>
+    <issn>2194-900X</issn>
+    <eissn>2509-839X</eissn>
+    <summary>Style for Annalen des Naturhistorischen Museums in Wien</summary>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: Fix summary please

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+          <name name-as-sort-order="all" and="symbol" font-variant="small-caps" initialize-with="." delimiter=", " delimiter-precedes-last="never"/>
+        </names>
+        <text macro="anon"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="author">
+    <names variable="author">
+      <name name-as-sort-order="all" and="symbol" sort-separator=", " initialize-with="." delimiter=", " delimiter-precedes-last="never" font-variant="small-caps"/>
+      <label form="short" prefix=" (" suffix=")" strip-periods="true"/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: you probably don't want `strip-periods="true"` here if you go through the trouble of defining ed. in the terms.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+      </if>
+    </choose>
+  </macro>
+  <macro name="container-contributors">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <names variable="editor translator" delimiter=", ">
+          <name name-as-sort-order="all" and="symbol" font-variant="small-caps" initialize-with="." delimiter=", " delimiter-precedes-last="never"/>
+        </names>
+        <text macro="anon"/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: Are we sure we want this here? I could see chapters in a book of the same author being cited occasionally -- we wouldn't want those to be labeled as anonymous.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        <single>ed.</single>
+        <multiple>eds</multiple>
+      </term>
+      <term name="no date">undated</term>
+    </terms>
+  </locale>
+  <macro name="anon">
+    <choose>
+      <if variable="author editor translator" match="none">
+        <text term="anonymous" text-case="capitalize-first"/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: and we're sure this isn't getting abbreviated? "Anonymous" seems very unusual compared to "Anon."

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon"/>
+  </macro>
+  <macro name="access">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper bill book chapter graphic legal_case legislation motion_picture paper-conference report song" match="none">
+        <text variable="container-title" font-style="italic" suffix=". " prefix=" "/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: remove the suffix here, the prefix for the DOI, and instead handle all of it with a group delimiter. (This, e.g. will print ". . " for items with container title and DOI.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    <choose>
+      <if variable="DOI">
+        <text variable="DOI" prefix=". https://doi.org/"/>
+      </if>
+      <else>
+        <choose>
+          <if type="webpage">
+            <group>
+              <text term="retrieved" text-case="capitalize-first" suffix=" "/>
+              <date variable="accessed" suffix=", ">
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: you have a duplicate space here with the prefix of `from`

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+              </group>
+            </group>
+          </if>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <group prefix=" ">
+      <text variable="genre" suffix=": " text-case="capitalize-first"/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: duplicate space with prefix on publisher. Why not handle this with a delimiter.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+      <choose>
+        <if type="book">
+          <text macro="edition"/>
+        </if>
+      </choose>
+    </group>
+  </macro>
+  <macro name="container-title">
+    <choose>
+      <if type="chapter">
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: normally chapter and paper-conference go together?

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+          <label plural="always" strip-periods="false" variable="page" form="short"/>
+          <text variable="number-of-pages"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="locators">
+    <choose>
+      <if type="article-journal article-magazine article-newspaper" match="any">
+        <text variable="container-title" font-style="normal"/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: I'd remove all the `normal` attributes. They're just clutter and we have occasionally seen them cause problems.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+        <text variable="volume" font-weight="normal" prefix=", "/>
+        <text variable="issue" prefix=" (" suffix=")"/>
+        <text variable="page" prefix=": "/>
+      </if>
+      <else-if type="bill book chapter graphic legal_case legislation motion_picture paper-conference report song" match="any">
+        <group>
+          <text term="in" suffix=": " text-case="capitalize-first"/>
+          <text macro="container-contributors" suffix=": "/>
+          <text macro="container-title" suffix="."/>
+          <label variable="page" form="short" suffix=". " prefix=" – "/>
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: no need for the period here. That's part of the term.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+  <bibliography hanging-indent="true" entry-spacing="1" line-spacing="1">
+    <sort>
+      <key macro="author"/>
+      <key variable="issued"/>
+    </sort>
+    <layout suffix=".">
+      <text macro="author"/>
+      <text macro="issued" prefix=" (" suffix="): "/>
+      <text macro="title" suffix=". – "/>
+      <group delimiter=". " suffix=" ">
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: don't use groups with a single element. They don't do anything. Move the element out and apply the suffix to it directly.

File: paleontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+                </group>
+              </group>
+            </if>
+          </choose>
+        </else>
+      </choose>
+    </group>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=" " prefix=" ">
```
Comment (adam3smith) [inline paleontographica-abteilung-b-palaeobotany-palaeophytology.csl]: This isn't quite right yet, I think. How is genre supposed to work here? Do you have an example?
E.g. Ph.D. Thesis: (Fancy University Smart Town) seems odd? 
And definitely no punctuation between publisher & place?

**Comment by adam3smith** [comment]
Apart from the remaining question in the publisher macro, looks good.

**Comment by rmzelle** [comment]
https://www.researchgate.net/publication/312218815_A_remarkable_extinct_wood_from_Lusatia_central_Europe_-_Juniperoxylon_schneiderianum_sp_nov_with_affinity_to_Cupressospermum_saxonicum_Mai has a reasonably recent full-text article, by the way.

File: palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+      <text variable="genre" suffix=": " text-case="capitalize-first"/>
+      <choose>
+        <if type="speech paper-conference" match="none">
+          <group delimiter=" ">
+            <text variable="publisher" prefix="(" suffix=")"/>
+            <text variable="publisher-place"/>
+          </group>
+        </if>
+      </choose>
+    </group>
```
Comment (rmzelle) [inline palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl]: @damnation333, @adam3smith, the relevant examples at https://www.schweizerbart.de/journals/palb/instructions:

> TAYLOR, T. N. (1981): Paleobotany: An introduction to fossil plant biology. – pp. 1–589 (McGraw-Hill Book Co.) New York.
> 
> FISCHER, O. (1990): Blätterfloren aus mitteleozänen Sedimenten des südlichen Weißelster-Beckens (Profen und Scheiplitz). – pp. 1–142, unpublished PhD thesis, Humboldt University Berlin.

You would store the "unpublished PhD thesis" in the "genre" CSL variable, right? In that case this doesn't look good yet.

File: palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="number-of-pages">
+    <choose>
+      <if type="book report" match="any">
+        <group delimiter=" ">
+          <label plural="always" strip-periods="false" variable="page" form="short"/>
+          <text variable="number-of-pages"/>
```
Comment (adam3smith) [inline palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl]: Given these odd instructions:

> TAYLOR, T. N. (1981): Paleobotany: An introduction to fossil plant biology. – pp. 1–589 (McGraw-Hill Book Co.) New York.
> 
> FISCHER, O. (1990): Blätterfloren aus mitteleozänen Sedimenten des südlichen Weißelster-Beckens (Profen und Scheiplitz). – pp. 1–142, unpublished PhD thesis, Humboldt University Berlin.

this should a) include thesis in line 192 and b) have `prefix="1&#8211;"` right?

File: palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+        </else>
+      </choose>
+    </group>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=" ">
+      <choose>
+        <if type="speech paper-conference" match="none">
+          <choose>
+            <if type="book chapter" match="any">
```
Comment (adam3smith) [inline palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl]: do we want to include report here, too?

File: palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+            </if>
+          </choose>
+        </else>
+      </choose>
+    </group>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=" ">
+      <choose>
+        <if type="speech paper-conference" match="none">
```
Comment (adam3smith) [inline palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl]: since you now test for all relevant item types, this isn't needed anymore.

**Comment by adam3smith** [comment]
Great, thanks!

**Pull request merged**